src/libraries/System.Threading.Channels/src/System/Threading/Channels/AsyncOperation.netstandard2.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Threading.Channels
+{
+    internal partial class AsyncOperation<TResult>
+    {
+        private void UnsafeQueueSetCompletionAndInvokeContinuation() =>
+            ThreadPool.UnsafeQueueUserWorkItem(static s => ((AsyncOperation<TResult>)s).SetCompletionAndInvokeContinuation(), this);
+
+        private static void QueueUserWorkItem(Action<object?> action, object? state) =>
+            ThreadPool.QueueUserWorkItem(new WaitCallback(action), state);
```

Comment:
Have you measured reverting this QueueUserWorkItem back to using Task.Factory.StartNew?  I suspect the previous code for this is a tad better.

Reply:
[QueueUserWorkItemOptimizationBenchmarks.zip](https://github.com/dotnet/runtime/files/5835206/QueueUserWorkItemOptimizationBenchmarks.zip)

The performance with and without change is quite the same with threshold 4%, but I found very small differences in allocations.

| Method | with | without |
| --------- |-----:|---------:|
| BoundedChannelPerfTests.PingPong | 645,197 B | 635,464 B |
| BoundedChannelPerfTests.PingPongWithExecutionContexts | 652,358 B | 667,669 B |
| SpscUnboundedChannelPerfTests.PingPong | 641,473 B | 633,981 B |
| SpscUnboundedChannelPerfTests.PingPongWithExecutionContexts | 645,224 B | 666,235 B |
| UnboundedChannelPerfTests.PingPong | 647,304 B | 637,213 B |
| UnboundedChannelPerfTests.PingPongWithExecutionContexts | 651,749 B | 674,043 B |

Task.Factory.StartNew is better with default ExecutionContext and worse with custom. PingPong has 10 000 iterations, so we have a difference of 1-3 bytes per iteration. It's hard for me to say which implementation is better. @stephentoub, what do you think?

Reply:
> @stephentoub, what do you think?

Let's stick with Task.Factory.StartNew.  On .NET Framework, modifying ExecutionContext means that every subsequent flowing incurs allocation, so most components then optimize for the non-modified case, including Task/StartNew.  This can do the same.

Reply:
Done.